dotnet/src/Connectors/Connectors.OpenAI/ToolCallBehavior.cs

```diff
@@ -35,6 +37,12 @@ public abstract class ToolCallBehavior
     /// </remarks>
     private const int DefaultMaximumAutoInvokeAttempts = 5;
 
+    /// <summary>
+    /// Gets the collection of filters that will be applied to tool calls.
+    /// </summary>
+    [Experimental("SKEXP0016")]
+    public IList<IToolFilter> Filters { get; } = new List<IToolFilter>();
```

Comment:
I don't think we want this here, at least not as a mutable list. It means any code can just do `ToolCallBehavior.AutoInvokeKernelFunctions.Filters.Add(myFilter)`, and it'll be added to the singleton that will apply to everyone, which also means you need to remember to remove filters after you're done with them, even in the case of exception.

I think instead we should add overloads to the existing factories below, e.g.
```C#
public static ToolCallBehavior EnableFunctions(
    IEnumerable<OpenAIFunction>? functions, // if null, functions are sourced from the Kernel ala AutoInvokeKernelFunctions,
    EnableFunctionsOptions options);
...
public sealed class EnableFunctionsOptions
{
    public bool AutoInvoke { get; set; }
    public IList<IToolFilter> Filters { get; }
    ... // any other customization desired
}
```
or something along those lines.  That's just a sketch; names and overall shape are debatable.